Backend/Controllers/GroupMemberships.cs

```
@@ -0,0 +1,167 @@
+    {
+        var cm = await db.GroupMemberships
+            .Include(e => e.Member)
+            .Include(e => e.Group)
+            .FirstOrDefaultAsync(e => e.Id == id, cancellationToken);
+
+        if (cm is null) return NotFound();
+
+        var result = new GroupMembershipResponseDTO
+        {
+            Id = cm.Id,
+            MemberId = cm.MemberId,
+            MemberName = $"{cm.Member.FirstName} {cm.Member.LastName}",
+            GroupId = cm.GroupId,
+            GroupName = cm.Group.Name,
+            GroupType = cm.Group.Type,
+            MembershipYear = cm.MembershipYear,
+            RoleId = cm.Role?.Id,
+            RoleName = cm.Role?.Name
+        };
+
+        return Ok(result);
+    }
+
+    // POST: api/groupMemberships
+    /// <summary>
+    /// Creates a new group membership with a unique ID assigned by the database.
+    /// </summary>
+    /// <param name="membershipDto">The group membership to be added to the database.</param>
+    /// <returns>Fully created group membership in body and api route of where to fetch it in the headers.</returns>
+    [HttpPost]
+    public async Task<ActionResult<GroupMembership>> PostGroupMembership(PostGroupMembershipDTO membershipDto, CancellationToken cancellationToken)
+    {
+        Member? member = await db.Members.FindAsync(membershipDto.MemberId, cancellationToken);
+        if (member is null)
+            return BadRequest($"Member with ID {membershipDto.MemberId} does not exist.");
+
+        Group? group = await db.Groups.FindAsync(membershipDto.GroupId, cancellationToken);
+        if (group is null)
+            return BadRequest($"Group with ID {membershipDto.GroupId} does not exist.");
+
+        var newMembership = new GroupMembership
+        {
+            Member = member,
+            Group = group,
+            MembershipYear = membershipDto.MembershipYear,
+            RoleId = membershipDto.RoleId
```

Add a TODO note here, saying that we should validate input given and should not blindly accept any value. What if the roleID does not point to a valid role? Not important for now, but should certainly not be forgotten. This is a problem at multiple points, not just here. Or perhaps this should be a new issue.

---

Is this not automatically checked as it is a foreign key?

---

Yes, and then the thread crashes :) You are telling EF Core to do something which is illegal. If, instead, you check if the input is valid you can return a 400 error instead.

---

Yeah fair point